istioctl/pkg/precheck/precheck_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package precheck
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"istio.io/istio/istioctl/pkg/cli"
+	"istio.io/istio/pkg/config/analysis/diag"
+	"istio.io/istio/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+type testCase struct {
+	version        string
+	revision       string
+	expectedOutput diag.Messages
+	expectedError  error
+}
+
+func Test_checkFromVersion(t *testing.T) {
+	cases := []testCase{
+		{
+			revision:       "canary",
+			version:        "1.21",
+			expectedOutput: diag.Messages{},
+			expectedError:  nil,
+		},
+		{
+			revision:       "canary",
+			version:        "XX",
+			expectedOutput: nil,
+			expectedError:  fmt.Errorf("invalid version XX, expected format like '1.0'"),
+		},
+		{
+			revision:       "canary",
+			version:        "2.0",
+			expectedOutput: nil,
+			expectedError:  fmt.Errorf("expected major version 1, got 2"),
+		},
+		{
+			revision:       "canary",
+			version:        "2.",
+			expectedOutput: nil,
+			expectedError:  fmt.Errorf("minor version is not a number: X"),
+		},
+		// TODO: add more test cases
```

Comment:
Just tbc, is this a TODO for now or for another PR?

Reply:
The plan is to add this in another PR